src/psyclone/tests/psyad/tl2ad_test.py

```diff
@@ -355,6 +355,60 @@ def test_generate_adjoint_kind(fortran_reader):
     assert ad_fortran_str in expected_ad_fortran_str
 
 
+def test_generate_adjoint_multi_kernel(fortran_writer):
+    '''Check that generate_adjoint creates the expected code when there
+    are multiple kernels in a module.
+
+    '''
+    expected = (
+        "module test_mod\n"
+        "  implicit none\n"
+        "  public\n\n"
+        "  contains\n"
+        "  subroutine adj_kern1()\n"
+        "    real :: psyir_tmp\n"
+        "    real :: psyir_tmp_1\n\n"
+        "    psyir_tmp = 0.0\n"
+        "    psyir_tmp_1 = 0.0\n"
+        "    psyir_tmp_1 = psyir_tmp_1 + psyir_tmp\n"
+        "    psyir_tmp = 0.0\n\n"
+        "  end subroutine adj_kern1\n"
+        "  subroutine adj_kern2()\n"
+        "    real :: psyir_tmp\n"
+        "    real :: psyir_tmp_1\n\n"
+        "    psyir_tmp = 0.0\n"
+        "    psyir_tmp_1 = 0.0\n"
+        "    psyir_tmp_1 = psyir_tmp_1 + psyir_tmp\n"
+        "    psyir_tmp = 0.0\n\n"
+        "  end subroutine adj_kern2\n"
+        "  subroutine adj_kern3()\n"
+        "    real :: psyir_tmp\n"
+        "    real :: psyir_tmp_1\n\n"
+        "    psyir_tmp = 0.0\n"
+        "    psyir_tmp_1 = 0.0\n"
+        "    psyir_tmp_1 = psyir_tmp_1 + psyir_tmp\n"
+        "    psyir_tmp = 0.0\n\n"
+        "  end subroutine adj_kern3\n\n"
+        "end module test_mod\n")
+    psyir = FileContainer("test_file")
+    kern_list = []
+    for name in ["kern1", "kern2", "kern3"]:
+        routine_symbol_table = SymbolTable()
```

Comment:
It's a minor point but it seems odd to construct the input code like this but have the expected output specified in monolithic fashion. Could you construct the Fortran for each kernel in this loop too?

Reply:
I've made it all monolithic as it was simpler. I hope that is OK.

Reply:
And it tests the full module rather than testing each routine at a time.

Reply:
OK, thanks.